File: frontend/src/types/forecasting.ts
export interface MetricProjectionResponse {
  metric: string
  current_value: number
  projected_value: number
  projected_change: number
  daily_change: number
  horizon_days: number
}

export interface FinancialForecastResponse {
  generated_at: string
  horizon_days: number
  history_points: number
  net_worth: MetricProjectionResponse
  cash_flow: MetricProjectionResponse
  account_balance: MetricProjectionResponse
  goal_progress: MetricProjectionResponse
  total_debt: MetricProjectionResponse
  health_score: MetricProjectionResponse
}

File: src/api/schemas/forecasting.py
"""API schemas for financial forecasting endpoints."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class MetricProjectionResponse(BaseModel):
    """Serialized representation of one projected financial metric."""

    model_config = ConfigDict(from_attributes=True)

    metric: str
    current_value: float
    projected_value: float
    projected_change: float
    daily_change: float
    horizon_days: int


class FinancialForecastResponse(BaseModel):
    """Serialized representation of a complete financial forecast."""

    model_config = ConfigDict(from_attributes=True)

    generated_at: datetime
    horizon_days: int
    history_points: int
    net_worth: MetricProjectionResponse
    cash_flow: MetricProjectionResponse
    account_balance: MetricProjectionResponse
    goal_progress: MetricProjectionResponse
    total_debt: MetricProjectionResponse
    health_score: MetricProjectionResponse
